File: cpp_on_sea/adler32/swift/adler32.swift
import Foundation

func adler32Checksum(data: [UInt8]) -> UInt32 {
    let modAdler: UInt32 = 65521
    var a: UInt32 = 1
    var b: UInt32 = 0

    for byte in data {
        a = (a + UInt32(byte)) % modAdler
        b = (b + a) % modAdler
    }

    return (b << 16) | a
}

// Generate or load 50 million bytes of data
let dataCount = 50_000_000
var byteArray = [UInt8]() 
byteArray.reserveCapacity(dataCount)
for _ in 0..<dataCount {
    byteArray.append(UInt8.random(in: 0...255))
}

// Compute Adler-32
let checksum = adler32Checksum(data: byteArray)
print(String(format: "Adler-32 Checksum: 0x%08X", checksum))

File: cpp_on_sea/adler32/kotlin/adler32.kt
import java.util.Random

fun main() {
    // Number of random bytes to generate
    val count = 50_000_000

    // Adler-32 parameters
    val MOD_ADLER = 65521
    var a = 1
    var b = 0

    // Initialize random generator
    val random = Random()

    // Generate and store random bytes
    val data = IntArray(count) { random.nextInt(256) }

    // Compute Adler-32 checksum
    for (byteValue in data) {
        a = (a + byteValue) % MOD_ADLER
        b = (b + a) % MOD_ADLER
    }

    // Combine sums into checksum
    val checksum = (b shl 16) or a
    println("Adler-32 checksum = $checksum")
}